src/ga4gh/vrs/_internal/models.py

```diff
@@ -133,120 +133,332 @@ def pydantic_class_refatt_map():
         ...,
-        description='A character representing a specific residue (i.e., molecular species) or groupings of these ("ambiguity codes"), using [one-letter IUPAC abbreviations](https://en.wikipedia.org/wiki/International_Union_of_Pure_and_Applied_Chemistry#Amino_acid_and_nucleotide_base_codes) for nucleic acids and amino acids.',
+        description='Indicates that the value is taken from a set of controlled strings defined elsewhere. Technically, a code is restricted to a string which has at least one character and no leading or  trailing whitespace, and where there is no whitespace other than single spaces in the contents.',
+        example='ENSG00000139618',
     )
 
 
-class SequenceString(RootModel):
-    root: constr(pattern=r'^[A-Z*\-]*$') = Field(
+class IRI(RootModel):
+    root: str = Field(
         ...,
-        description='A character string of Residues that represents a biological sequence using the conventional sequence order (5’-to-3’ for nucleic acid sequences, and amino-to-carboxyl for amino acid sequences). IUPAC ambiguity codes are permitted in Sequence Strings.',
+        description='An IRI Reference (either an IRI or a relative-reference), according to `RFC3986 section 4.1  <https://datatracker.ietf.org/doc/html/rfc3986#section-4.1>` and `RFC3987 section 2.1 <https://datatracker.ietf.org/doc/html/rfc3987#section-2.1>`. MAY be a JSON Pointer as an IRI fragment, as  described by `RFC6901 section 6 <https://datatracker.ietf.org/doc/html/rfc6901#section-6>`.',
     )
 
 
-class MappingClass(Enum):
-    closeMatch = 'closeMatch'
-    exactMatch = 'exactMatch'
-    broadMatch = 'broadMatch'
-    narrowMatch = 'narrowMatch'
-    relatedMatch = 'relatedMatch'
-
-
 class Extension(BaseModel):
+    """The Extension class provides VODs with a means to extend descriptions with other
+    attributes unique to a content provider. These extensions are not expected to be
+    natively understood under VRSATILE, but may be used for pre-negotiated exchange of
+    message attributes when needed.
+    """
     model_config = ConfigDict(
         extra='allow',
     )
-    type: str = Field('Extension', description='MUST be "Extension".')
+    type: Literal['Extension'] = Field('Extension', description='MUST be "Extension".')
     name: str = Field(..., description='A name for the Extension')
     value: Optional[Union[float, str, bool, Dict[str, Any], List[Any]]] = Field(
         None, description='Any primitive or structured object'
     )
 
 
-class Code(RootModel):
-    root: constr(pattern=r'\S+( \S+)*') = Field(
-        ...,
-        description='Indicates that the value is taken from a set of controlled strings defined elsewhere. Technically, a code is restricted to a string which has at least one character and no leading or  trailing whitespace, and where there is no whitespace other than single spaces in the contents.',
-        example='ENSG00000139618',
+class _Entity(BaseModel):
```

Comment:
Is there a difference in meaning between classes beginning with an underscore and those not? Some seem kind of like 'internal use' like `_Ga4ghIdentifiableObject`, but others like `DomainEntity` are in the model sources (but not the processed jsonschemas, since they're defined as 'abstract'(?) in the metaschema source files?).

Reply:
@theferrit32 Correct, they're meant to be internal. I also realized that maybe I should move the core models to `src.ga4gh.core._internal.models.py`

Reply:
@theferrit32 pushed change to separate out core models. 

```markdown
  * `from ga4gh.core import core_models`, and refer to models with the
    abbreviated name, e.g., `core_models.Gene` (recommended)

  * `import ga4gh.core`, and refer to models using the fully-qualified
    module name, e.g., `ga4gh.core.core_models.Gene`
```